src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -13,6 +14,8 @@ public class RequestServicesContainerMiddleware
         private readonly RequestDelegate _next;
         private readonly IServiceProvider _services;
 
+        private IServiceScopeFactory _scopeFactory;
+
         public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider services)
```

Comment:
Why not just pass it in here? It's eager sure, but it's just one resolve. You get to drop the rest of the complexity and just pass it in a parameter to the feature.